sdk/monitor/monitor-query/src/logsQueryClient.ts

```diff
@@ -180,6 +180,84 @@ export class LogsQueryClient {
       return result;
     });
   }
+
+  /**
+   * Queries logs in a Log Analytics Resource.
+   *
+   * @param resourceId - The 'Resource Id' for the Log Analytics Resource
+   * @param query - A Kusto query.
+   * @param timespan - The timespan over which to query data. This is an ISO8601 time period value. This timespan is applied in addition to any that are specified in the query expression.
+   *  Some common durations can be found in the `Durations` object.
+   * @param options - Options to adjust various aspects of the request.
+   * @returns The result of the query.
+   */
+  async queryResource(
+    resourceId: string,
+    query: string,
+    timespan: QueryTimeInterval,
+    // eslint-disable-next-line @azure/azure-sdk/ts-naming-options
+    options: LogsQueryOptions = {}
+  ): Promise<LogsQueryResult> {
+    let timeInterval: string = "";
+    return tracingClient.withSpan(
+      "LogsQueryClient.queryResource",
+      options,
+      async (updatedOptions) => {
+        if (timespan) {
+          timeInterval = convertTimespanToInterval(timespan);
+        }
+        const { flatResponse, rawResponse } = await getRawResponse(
+          (paramOptions) =>
+            this._logAnalytics.query.resourceExecute(
+              resourceId,
```

Comment:
When the resourceId is copied from portal, it usually begins with `/resource/...`. We should trim the `/` prefix (if it exists) from user input to ensure the request URL does not contain `//`.

Reply:
updated